File: appiumtests/helpers/stub_gamescope.py
#!/usr/bin/env python3
# SPDX-License-Identifier: GPL-3.0-or-later
# SPDX-FileCopyrightText: 2025 CouchPlay Contributors
#
# Stub "gamescope": opens a window whose Wayland app_id / WM_CLASS is "gamescope"
# so the app's WindowManager (which matches windows by resourceClass ==
# "gamescope") can find and position it. Stays open until killed. Anything after
# `--` (the game command) is ignored.
#
# Used by mock_helper.LaunchInstance when COUCHPLAY_STUB_GAMESCOPE points here.
#
# NOTE: we deliberately do NOT use Gtk.Application — its application_id must be
# reverse-DNS, which would not yield resourceClass "gamescope". Instead we set
# the program name + WM_CLASS so GTK's Wayland backend reports app_id "gamescope".
import sys

import gi

gi.require_version("Gtk", "3.0")
from gi.repository import GLib, Gtk

GLib.set_prgname("gamescope")

win = Gtk.Window(title="gamescope")
win.set_wmclass("gamescope", "gamescope")
win.set_default_size(512, 768)
win.connect("destroy", Gtk.main_quit)
win.show_all()

Gtk.main()

File: appiumtests/helpers/stub_gamescope.cpp
// SPDX-License-Identifier: GPL-3.0-or-later
// SPDX-FileCopyrightText: 2025 CouchPlay Contributors
//
// Stub "gamescope" window for e2e session-lifecycle tests. Compiled to a binary
// NAMED "gamescope" so Qt6 Wayland reports app_id/resourceClass "gamescope",
// which the app's WindowManager (findAllGamescopeWindows) matches. A Python/
// PySide6 stub does NOT work here -- Qt6 Wayland takes the app_id from the
// executable name, which would be "python3".
//
// Built by appiumtests/Dockerfile to /opt/couchplay-e2e/gamescope and launched
// by mock_helper.LaunchInstance (COUCHPLAY_STUB_GAMESCOPE).
#include <QApplication>
#include <QWidget>

int main(int argc, char **argv)
{
    QApplication app(argc, argv);
    app.setApplicationName("gamescope");

    QWidget window;
    window.setWindowTitle("gamescope");
    window.resize(512, 768);
    window.show();

    return app.exec();
}
